modules/apps/commerce/commerce-checkout-web/src/main/java/com/liferay/commerce/checkout/web/internal/helper/DefaultCommerceCheckoutStepHttpHelper.java

```diff
@@ -666,9 +720,11 @@ private CommerceShippingOption _getSingleCommerceShippingOption(
 					commerceShippingMethod.getEngineKey());
 
 			List<CommerceShippingOption> commerceShippingOptions =
-				commerceShippingEngine.getCommerceShippingOptions(
-					commerceContext, commerceOrder,
-					_portal.getLocale(httpServletRequest));
+				_getFilteredCommerceShippingOptions(
```

Comment:
I'm worried this will not work. Not all shipping options are of type FixedShippingOption, for example Fedex options aren't. So if we want to implement this filter it must be in the Shipping engine related to that method.

Reply:
@luca-pellizzon ahh got it. Looks like there was already a method implemented haha. Thanks for catching that!